plugin/trino-elasticsearch/src/test/java/io/trino/plugin/elasticsearch/BaseElasticsearchConnectorTest.java

```diff
@@ -459,6 +461,292 @@ public void testArrayFields()
+                "es_array_object[1].array_of_int_arrays " +
+                "FROM raw_json " +
+                "ORDER BY order_field");
+
+        MaterializedResult arrayExpected = resultBuilder(getSession(), arrayRows.getTypes())
+                .row("\"abc\"", "\"def\"", "[[\"abc\",\"def\"]]", "123", "234", "345", "[123,[234,345]]")
+                .row(null, null, "\"If you like Presto, you'll love Trino: https://trino.io/slack.html\"", null, null, null, "321")
+                .build();
+
+        assertEquals(arrayRows.getMaterializedRows(), arrayExpected.getMaterializedRows());
+
+        MaterializedResult rawRows = computeActual("" +
+                "SELECT " +
+                "json_extract(es_raw_object, '$.array_of_string_arrays[0][0]'), " +
+                "json_extract(es_raw_object, '$.array_of_string_arrays[0][1]'), " +
+                "json_extract(es_raw_object, '$.array_of_string_arrays'), " +
+                "json_extract(es_raw_object, '$.array_of_int_arrays[0]'), " +
+                "try(json_extract(es_raw_object, '$.array_of_int_arrays[1][0]')), " +
+                "try(json_extract(es_raw_object, '$.array_of_int_arrays[1][1]')), " +
+                "json_extract(es_raw_object, '$.array_of_int_arrays') " +
+                "FROM raw_json " +
+                "ORDER BY order_field");
+
+        MaterializedResult rawRowsExpected = resultBuilder(getSession(), rawRows.getTypes())
+                .row("\"abc\"", "\"def\"", "[[\"abc\",\"def\"]]", "123", "234", "345", "[123,[234,345]]")
+                .row(null, null, "\"The founders and core contributors of Presto, and are now working on Trino: https://trino.io/blog/2020/12/27/announcing-trino.html\"", null, null, null, "654")
+                .build();
+
+        assertEquals(rawRows.getMaterializedRows(), rawRowsExpected.getMaterializedRows());
+    }
+
+    @Test
+    public void testAsRawJsonAndIsArraySameFieldException()
+            throws IOException
+    {
+        String indexName = "raw_json_array_exception";
+
+        @Language("JSON")
+        String mapping = "" +
+                "{" +
+                "  \"_meta\": {" +
+                "    \"presto\": {" +
+                "      \"array_raw_field\": {" +
+                "        \"asRawJson\": true," +
+                "        \"isArray\": true" +
+                "      }" +
+                "    }" +
+                "  }," +
+                "  \"properties\": {" +
+                "    \"array_raw_field\": {" +
+                "      \"type\": \"text\"" +
+                "    }" +
+                "  }" +
+                "}";
+
+        createIndex(indexName, mapping);
+
+        index(indexName, ImmutableMap.<String, Object>builder()
+                .put("array_raw_field", "test")
+                .build());
+
+        assertThatThrownBy(() -> computeActual("SELECT array_raw_field FROM raw_json_array_exception"))
```

Comment:
Let's assert the instance of exception thrown as well. We need to avoid any NPE or any RTE thrown from passing this test.